src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,54 +1,100 @@
-      <div className="field">
-        <label className="label" htmlFor="movie-title">
-          Movie title
-        </label>
-
-        <div className="control">
-          <input
-            type="text"
-            id="movie-title"
-            placeholder="Enter a title to search"
-            className="input is-danger"
-          />
-        </div>
+interface Props {
+  callbackForMovies: (arg0 : Movie[]) => void;
+  moviesPrev: Movie[];
+}
 
-        <p className="help is-danger">
-          Can&apos;t find a movie with such a title
-        </p>
-      </div>
+export const FindMovie: React.FC<Props> = (
+  { callbackForMovies, moviesPrev },
+) => {
+  const [movieForFind, setMovieForFind] = useState('');
+  const [preview, setPreview] = useState<Movie>();
+  const [isFilmValid, setIsFilmValid] = useState(true);
+
+  async function finder() {
+    const result = await request(movieForFind.toLowerCase());
+
+    if (result.Response !== 'False') {
+      setPreview(result);
+    } else {
+      setMovieForFind('');
+      setPreview(undefined);
+      setIsFilmValid(false);
+    }
+  }
 
-      <div className="field is-grouped">
-        <div className="control">
-          <button
-            type="button"
-            className="button is-light"
-          >
-            Find a movie
-          </button>
+  return (
+    <>
+      <form className="find-movie">
+        <div className="field">
+          <label className="label" htmlFor="movie-title">
+            Movie title
+          </label>
+
+          <div className="control">
+            <input
+              type="text"
+              id="movie-title"
+              placeholder="Enter a title to search"
+              value={movieForFind}
+              className={classNames('input', { 'is-danger': !isFilmValid })}
+              onChange={(event) => {
+                setMovieForFind(event.target.value);
+              }}
+            />
+          </div>
+
+          { !isFilmValid
+          && (
+            <p className="help is-danger">
+              Can&apos;t find a movie with such a title
+            </p>
+          )}
         </div>
 
-        <div className="control">
-          <button
-            type="button"
-            className="button is-primary"
-          >
-            Add to the list
-          </button>
+        <div className="field is-grouped">
+          <div className="control">
+            <button
+              type="button"
+              className="button is-light"
+              onClick={() => {
+                finder();
+                setMovieForFind('');
+              }}
+            >
+              Find a movie
+            </button>
+          </div>
+
+          <div className="control">
+            <button
+              type="button"
+              className="button is-primary"
+              onClick={() => {
+                if (preview) {
+                  if (!moviesPrev.find(el => el.imdbID === preview.imdbID)) {
```

Comment:
it is better to do this in App.tsx, so FindMovie component receive only function

Reply:
agree, move this check to App.tsx component